libs/vm-modules/tests/unit/fixed_point.cpp

```diff
@@ -237,4 +237,43 @@ TEST_F(FixedPointTest, multidigit_integral_fixed_point128)
   EXPECT_TRUE(toolkit.Run());
 }
 
+TEST_F(FixedPointTest, fixed_point128_shallow_copy)
+{
+  static constexpr char const *SOURCE = R"(
+      function main()
+        var a = 9876.54321000fp128;
+        var b = 9876.54321000fp128;
+
+        a = b;
+        assert(a == b, "shallow copy failed!");
+
+        a += 1.0fp128;
+
+        assert(a == b, "shallow copy failed!");
+      endfunction
+    )";
+
+  EXPECT_TRUE(toolkit.Compile(SOURCE));
+  EXPECT_TRUE(toolkit.Run());
+}
+
+TEST_F(FixedPointTest, fixed_point128_deep_copy)
+{
+  static constexpr char const *SOURCE = R"(
+      function main()
+        var a = 9876.54321000fp128;
+        var b = 9876.54321000fp128;
+
+        a = b.copy();
+        assert(a == b, "deep copy failed!");
+
+        a += 1.0fp128;
+
+        assert(a > b, "deep copy failed!");
```

Comment:
For consistency with the other test (whose message I understand more clearly):
```suggestion
        assert(a > b, "b is corrupted by increasing a!");
```